### ISSUE TITLE
position_ids with left padding

### ISSUE THREAD

**Issue by JaheimLee** [tloen/alpaca-lora#315]
Hi, I have a question. According to the [huggingface code](https://github.com/huggingface/transformers/blob/151425ddb29d4ad1a121e8cce62000a2ac52d3ba/src/transformers/models/llama/modeling_llama.py#L522), `position_ids` is from 0 to seq_length. But if the padding side is left, text  position doesn't start from 0. Will it cause performance drop?

**Comment by ChuanMeng** [comment]
I have the same concern.

**Comment by REIGN12** [comment]
I am also curious about this! Is there anyone have any conclusions on this?

**Comment by raphael-sch** [comment]
yes, pretty sure the position_ids are incorrect during training with left padding. You would need to write a custom DataCollator and "manually" create position_ids for each instance in your batch

**Comment by raphael-sch** [comment]
minimal example:
https://gist.github.com/raphael-sch/c38c4dbaecce62566bc3f07559678360
